libsweep/src/sweep.cc

```diff
@@ -100,33 +109,60 @@ void sweep_device_start_scanning(sweep_device_s device, sweep_error_s* error) {
   if (device->is_scanning)
     return;
 
-  sweep::protocol::error_s protocolerror = nullptr;
-  sweep::protocol::write_command(device->serial, sweep::protocol::DATA_ACQUISITION_START, &protocolerror);
-
-  if (protocolerror) {
-    *error = sweep_error_construct("unable to send start scanning command");
-    sweep::protocol::error_destruct(protocolerror);
+  // Get the current motor speed setting
+  sweep_error_s speederror = nullptr;
+  int32_t speed = sweep_device_get_motor_speed(device, &speederror);
+  if (speederror) {
+    *error = sweep_error_construct("unable to start scanning. could not verify motor speed.");
+    sweep_error_destruct(speederror);
     return;
   }
+  // Check that the motor is not stationary
+  if (speed == 0 /*Hz*/) {
+    // If the motor is stationary, adjust it to default 5Hz
+    sweep_device_set_motor_speed(device, 5 /*Hz*/, &speederror);
```

Comment:
Yes. How would you like to do so? The API should concern itself with speeds and sample rates etc... but on the implementation side, I think it would be best if we operate in terms of protocol codes (speed setting code, sample rate setting code, error code, status byte code etc). 

I'm not sure what the best container for these will be. It might be nice in certain circumstances to be able to reference a "DEFAULT_MOTOR_SPEED_CODE" or something, but in other cases it might be useful to loop through a container to check for associated info related to specific code (ie: to get the sample rate in HZ from a sample rate code).